react/react1/week3/todo-app/src/components/Todo.js

```diff
@@ -0,0 +1,82 @@
+import PropTypes from 'prop-types';
+
+import { useState, useContext } from 'react';
+import { ToDosContext } from '../contexts/ToDosContext';
+
+export default function Todo({ todo }) {
+  const [checked, setChecked] = useState(false);
+  const [update, setUpdate] = useState(false);
+  const [descriptionValue, setDescriptionValue] = useState(todo.description);
+
+  const { handleDelete, handleAddUpdate } = useContext(ToDosContext);
+
+  const handleUpdate = () => {
+    if (descriptionValue) {
+      handleAddUpdate({
```

Comment:
I would definitely recommend split the add and update clearly into two. Nothing super wring with combining and you did it quite cleanly, however, separation of concerns would make i more readable here 🙏🏽

Reply:
I refactored the function 😄